### PR TITLE
Fix material testing inaccuracies by using assert_allclose for value comparison

### PR DESCRIPTION
**Pull request by ahnaf-tahmid-chowdhury**
## Description: 

In the material tests on macOS, there were issues due to minor floating-point inaccuracies, as values were close but not exactly the same. To resolve this, I replaced strict equality checks with assert_allclose to allow for a small tolerance in value comparison, ensuring the tests pass with similar values. This approach improves the reliability of the tests across different environments where floating-point precision may vary slightly.

### CODE REVIEW

**Comment by gonuke** [comment]
Have you experimented to see how low we can go with `rtol` and still match?

File: tests/test_material.py [hunk 1]
```diff
@@ -2042,22 +2043,26 @@ def test_material_gammas():
...
+            (93.35, 7.135646069465136e-18),
+            (89.953, 4.4112564001433475e-18),
+            (105.0, 3.394789326064896e-18),
+            (13.0, 2.9652250902501514e-17),
+            (93.35, 5.2767097753586244e-21),
+            (89.953, 3.2620619831267015e-21),
+            (105.0, 2.510398896994686e-21),
+            (13.0, 3.4433858446293e-17),
+        ],
+        rtol=1e-10, atol=0
```
Comment (gonuke) [inline tests/test_material.py]: Can we set `rtol` once and use that variable everywhere?

**Review by gonuke — CHANGES_REQUESTED** [review]
Thanks @ahnaf-tahmid-chowdhury - just one question about setting a single value of `rtol`

File: tests/test_material.py (hunk 1 above)
Comment (ahnaf-tahmid-chowdhury) [reply tests/test_material.py]: Done

File: tests/test_material.py [hunk 2]
```diff
@@ -46,6 +47,11 @@
 
 leu = {922380000: 0.96, 922350000: 0.04}
 
+# Relative Tolerance
+rtol = 1e-10
```
Comment (gonuke) [inline tests/test_material.py]: Do we know if this works at 1e-12?

**Review by gonuke — CHANGES_REQUESTED** [review]

File: tests/test_material.py (hunk 2 above)
Comment (ahnaf-tahmid-chowdhury) [reply tests/test_material.py]: We can test this while working on PR #1549. For now I am setting the value to 1e-20.

**Review by gonuke — APPROVED** [review]
Thanks @ahnaf-tahmid-chowdhury

**Pull request merged**